src/MiddleMail/MiddleMailService.cs

```diff
@@ -14,17 +18,32 @@ namespace MiddleMail {
 	/// Implements a graceful shutdown by waiting for all processing tasks to finish work if cancellation is requested.
 	/// </summary>
 	public class MiddleMailService : BackgroundService {
+		public static readonly TimeSpan RateLimitWindow = TimeSpan.FromMinutes(1);
+		public static readonly TimeSpan RateLimitDelay = TimeSpan.FromMinutes(1);
 		
 		private IMessageProcessor processor;
 		private readonly ILogger<MiddleMailService> logger;
 		private readonly IMessageSource messageSource;
 		private int consumerTasksPending;
+		private readonly MiddleMailOptions options;
+
+		private readonly RateLimiter? rateLimiter;
 
-		public MiddleMailService(IMessageProcessor processor, ILogger<MiddleMailService> logger, IMessageSource messageSource) {
+		public MiddleMailService(IOptions<MiddleMailOptions> options, IMessageProcessor processor, ILogger<MiddleMailService> logger, IMessageSource messageSource) {
 			this.processor = processor;
 			this.logger = logger;
 			this.consumerTasksPending = 0;
 			this.messageSource = messageSource;
+			this.options = options.Value;
+
+			if (this.options.RateLimited) {
```

Comment:
For a better integration with DI you can have a look at https://github.com/dotnet/aspnetcore/blob/main/src/Middleware/RateLimiting/src/RateLimitingMiddleware.cs and https://github.com/dotnet/aspnetcore/blob/main/src/Middleware/RateLimiting/src/RateLimiterOptions.cs.

Reply:
If you do it like that, you are probably able to improve the tests significantly, because you can mock the rate limiter :)

Reply:
I took a look at the examples you linked, but I can't understand how `GlobalLimiter` gets assigned in `RateLimiterOptions`. However it's assigned in those examples, we need to have the value be populated from environment variables for this program. Do you know how to do that with something like an object (as opposed to a string)?